pkg/serverless/logs/logs_test.go

```diff
@@ -218,15 +234,15 @@ func TestProcessMessageValid(t *testing.T) {
 	mockExecutionContext := &executioncontext.ExecutionContext{}
 	mockExecutionContext.SetFromInvocation(arn, lastRequestID)
 
-	go processMessage(message, mockExecutionContext, computeEnhancedMetrics, metricTags, demux, func() {})
+	processMessage(&message, mockExecutionContext, computeEnhancedMetrics, metricTags, demux, func() {})
 
 	received, timed := demux.WaitForSamples(100 * time.Millisecond)
 	assert.Len(t, received, 7)
 	assert.Len(t, timed, 0)
 	demux.Reset()
 
 	computeEnhancedMetrics = false
-	go processMessage(message, mockExecutionContext, computeEnhancedMetrics, metricTags, demux, func() {})
+	processMessage(&message, mockExecutionContext, computeEnhancedMetrics, metricTags, demux, func() {})
```

Comment:
I needed to remove the`go` call here. Otherwise, there is a race condition in the tests. If `processMessage` is run in a separate goroutine, the field `message.stringRecord` is written in two different goroutines -- first when the `logMessage` is created in line 215, second when the `stringRecord` value is added in the call to `processMessage` in lines 237 and 245.

The best way to avoid this race condition is by removing the goroutines. This brings the test case more in line with how the extension works in the wild.

Reply:
Hmm, I think we might have placed this goroutine here in the test to simulate how the processMessage API could actually be called, (so it would trigger the race detector and expose the data race). Were you able to verify which goroutines the message could be modified from in the production code?

Reply:
That said, I do understand your point of wanting to give the race detector any extra hand we can in helping us find issues. So, I'm happy to continue calling `processMessage` in a separate goroutine if you think that's best.

Reply:
Your reasoning sounds solid, sounds like we can skip the goroutine in this test.